### ISSUE TITLE
[Improvement] Loss class/model class separation

### ISSUE THREAD

**Issue by nm19000** [schererant/uq-qnn#24]
> Why is the loss a class called class PhotonicModel(torch.nn.Module): in uq-qnn/src/loss.py? It outputs the gradients times g_out ? what is g_out  in uq-qnn/src/autograd.py? 

 _Originally posted by @nm19000 in [#19](https://github.com/schererant/uq-qnn/issues/19#issuecomment-3873155873)_

Separate loss computation and model classes and for PSR do grad computation with model as input.
